### ISSUE TITLE
E019 gives a false positive for `%~dp0`

### ISSUE THREAD

**Issue by grinser** [tboy1337/Blinter#35]
## Description

Blinter reports rule E019 for the variable `%~dp0`. This report is not correct.
`%0` is a command-line parameter. The percent-tilde syntax is valid with `%0`.

## How to reproduce the problem

1. Make a file with the name `test.bat`.
2. Put this text in the file:

```bat
@echo off
set "OUT=%~dp0log.txt"
exit /b 0
```

3. Run this command:

```
python -m blinter test.bat
```

## Result

Blinter shows this message:

```
Line 2: Percent-tilde on non-parameter variable (E019)
```

## Expected result

Blinter does not show rule E019 for `%~dp0`.

## Reason

`%0` contains the path of the batch file. The modifier `~d` gives the drive.
The modifier `~p` gives the path. Thus `%~dp0` gives the directory of the
script. This is the standard method to find the directory of a script.
Microsoft shows this syntax in the help text of the `for` command.

The message of rule E019 says:

> Percent-tilde syntax only works with command-line parameters (%1-%9) and FOR
> loop variables

This message does not include `%0`. But `%0` is also a command-line parameter.
Therefore the rule must accept `%0`.

## Effect on users

Scripts that find their own directory use `%~dp0`. This is a usual pattern.
Blinter shows an error for each of these scripts. The error has the severity
level ERROR. Thus the user can think that the script has a defect.

## Environment

| Item | Value |
|---|---|
| Blinter | 1.1.9 (from PyPI) |
| Python | 3.14.5 |
| Operating system | macOS 26.6 |

## Related issue

Issue #32 corrects a related false positive. That issue is for FOR loop
variables (`%%~zA`). This issue is for `%0`.

**Comment by tboy1337** [comment]
Thanks for the report, it should now be fixed in v1.1.10 https://github.com/tboy1337/Blinter/commit/ca35a2b49e545540eda4ab8c81dcf10a71416b1a, if there are more issues please let me know.
